File: org.envirocar.app/src/org/envirocar/app/view/tracklist/OnTrackInteractionCallback.java
package org.envirocar.app.view.tracklist;

import android.view.View;

import org.envirocar.app.storage.Track;

/**
 * @author dewall
 */
interface OnTrackInteractionCallback {

    /**
     *
     * @param track the track to show the details for.
     */
    void onTrackDetailsClicked(Track track, View transitionView);

    /**
     *
     * @param track the track to delete.
     */
    void onDeleteTrackClicked(Track track);

    /**
     *
     * @param track the track to upload.
     */
    void onUploadTrackClicked(Track track);

    /**
     *
     * @param track the track to export.
     */
    void onExportTrackClicked(Track track);
}

File: org.envirocar.app/src/org/envirocar/app/views/tracklist/OnTrackInteractionCallback.kt
package org.envirocar.app.views.tracklist

import android.view.View
import org.envirocar.core.entity.Track

interface OnTrackInteractionCallback {
    fun onTrackDetailsClicked(track: Track, transitionView: View)
    fun onDeleteTrackClicked(track: Track)
    fun onUploadTrackClicked(track: Track)
    fun onShareTrackClicked(track: Track)
    fun onDownloadTrackClicked(track: Track, holder: AbstractTrackListCardAdapter.TrackCardViewHolder)
    fun onLongPressedTrack(track: Track)
}
